docs/docs/dev_docs/testing/cheat_codes.md

```diff
@@ -0,0 +1,449 @@
+---
+title: Testing Cheat Codes
+---
+
+## Introduction
+
+Most of the time, simply testing your smart contract isn't enough. To manipulate the state of the Aztec blockchain, as well as test for specific reverts and events, the sandbox is shipped with a set of cheatcodes.
+
+Cheatcodes allow you to change the time of the L2 block, load certain state or more easily manipulate L1 instead of having to write dedicated RPC calls to anvil. 
+
+:::info Prerequisites
+If you aren't familiar with [Anvil](https://book.getfoundry.sh/anvil/), we recommend reading up on that since Aztec Sandbox uses Anvil as the local L1 instance.
+:::
+
+### Aims
+
+The guide will cover how to manipulate the state of the:
+- L1 blockchain
+- Aztec network.
+### Why is this useful?
+
+<!-- Contextualise why a dapp developer needs this. What use cases / products / features does this unlock? Any real world examples? -->
+
+### Dependencies
+
+For this guide, the following Aztec packages are used:
+- @aztec/aztec.js
+
+### Initialisation
+```js
+import { createAztecRpcClient, CheatCodes } from '@aztec/aztec.js';
+const aztecRpcUrl = 'http://localhost:8080';
+const aztecRpcClient = createAztecRpcClient(aztecRpcUrl);
+const cc = CheatCodes.create(aztecRpcUrl, aztecRpcClient);
+```
+
+There are two properties of the CheatCodes class - `l1` and `l2` for cheatcodes relating to the L1 and L2 (Aztec) respectively.
+
+## L1 related cheatcodes
+These are cheatcodes exposed from anvil. But instead of having to know the right RPC calls to make and format the curl request appropirately, there are convenient wrappers exposed to developers.
+
+### Interface
+```
+// Fetch current block number of L1
+public async blockNumber(): Promise<number> 
+
+// Fetch chain ID of L1
+public async chainId(): Promise<number> 
+
+// Fetch current timestamp on L1
+public async timestamp(): Promise<number> 
+
+// Mine a given number of blocks on L1. Mines 1 block by default
+public async mine(numberOfBlocks = 1): Promise<void> 
+
+// Set the timestamp for the next block on L1. 
+public async setNextBlockTimestamp(timestamp: number): Promise<void> 
+
+// Dumps the current L1 chain state to a given file. Can be used with `loadChainState()` to first create a forked version of mainnet on anvil (to for example interact with uniswap), save the chain state to a file, and later load it to the sandbox' anvil instance.
```

Comment:
Should probably be multiple lines, if seen in most browsers will move out of the field of view.